components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java

```diff
@@ -180,12 +180,12 @@ public void generateEntitlementPolicies(APIIdentifier apiIdentifier) throws
      * @param policyPartial     policy content
      * @param isSharedPartial   policy status
      * @param policyAuthor      author of the policy
-     * @param policyPartialDesc policy description
+     * @param policyPartialDescription policy description
      * @return policy id
      * @throws AppManagementException
      */
     int saveEntitlementPolicyPartial(String policyPartialName, String policyPartial, boolean isSharedPartial,
-                                     String policyAuthor,String policyPartialDesc) throws AppManagementException;
+                                     String policyAuthor,String policyPartialDescription) throws AppManagementException;
```

Comment:
Consider introducing a bean to wrap all these pieces of information and then pass it as an argument into the method to be able to better organize the code (particularly, since all these input parameters seem to have some sort of a correlation with each other).